Update source setup - Closes #642 (#654)
LiskHQ/lisk-docs: modules/ROOT/pages/setup/source.adoc

--
====

=== PostgreSQL

To install Postgres follow the instructions.
Two alternative installation methods are explained below.

[tabs]
====
Postgres with Docker::
+
--
Running Postgres inside a Docker container will setup the correct version of Postgres and containerize it away from any existing versions that may exist locally on your machine.
Choose this setup if you are unfamiliar with Postgres, or if other issues occur with a previously installed version of Postgres.
To successfully perform the command shown below, install Docker as described in the Setup page of the xref:{url_docker_setup][Lisk Core Docker image].

> WARNING: If other versions of PostgreSQL are currently installed on your machine, please ensure they are stopped before starting the docker container.

Reviewer: ```suggestion
WARNING: If other versions of PostgreSQL are currently installed on your machine, please ensure they are stopped before starting the Docker container.
```

[source,bash]
----
docker run --name lisk_service_db -p 5432:5432 -e POSTGRES_USER=lisk -e POSTGRES_PASSWORD=password -e POSTGRES_DB=lisk -d postgres:10
----

This will install PostgreSQL version 10 (`postgres:10`) in a container with name `lisk_service_db` and binds the port `5432` of the container with the same port of the machine.

> As environment variables `POSTGRES_USER=lisk` is exposed in order to create the Lisk user, and `POSTGRES_PASSWORD=password` is used to set the password for the lisk user.

Reviewer: ```suggestion
As environment variables `POSTGRES_USER=lisk` is exposed in order to create the Lisk user, and `POSTGRES_PASSWORD=password` is used to set the password for the Lisk user.
```

> As environment variables `POSTGRES_USER=lisk` is exposed in order to create the Lisk user, and `POSTGRES_PASSWORD=password` is used to set the password for the lisk user.

Author: ```suggestion
As environment variables `POSTGRES_USER=lisk` is exposed in order to create the Lisk user, and `POSTGRES_PASSWORD=password` is used to set the password for the `lisk` user.
```

Finally the environment variable `POSTGRES_DB` creates the database `lisk` with the `lisk` user as the owner.

The above commands should be enough to set up the database ready to use with Lisk Service.
To manage the Docker container, use the following commands displayed below:

[source,bash]
----
docker stop lisk_service_db <1>
docker start lisk_service_db <2>
docker restart lisk_service_db <3>
docker rm lisk_service_db <4>
----

<1> Stop the container.
<2> Start the container.
<3> Restart the container.
<4> Remove the container.

> In order to access Postgres inside the container via CLI, run the following command displayed below:

Reviewer: ```suggestion
In order to access Postgres inside the container via the CLI, execute the following command displayed below:
```

----
docker exec --tty --interactive lisk_core_db psql -h localhost -U lisk -d postgres
----
--
Postgres system wide::
+
--
*Ubuntu*

Firstly, install PostgreSQL on your machine as shown in the following commands listed below:

[source,bash]
----
sudo apt-get purge -y postgres* <1>
sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
sudo apt install wget ca-certificates
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
sudo apt update
sudo apt install postgresql-10
----

<1> Remove all previously installed postgres versions.

After the installation is completed it should be possible to see the Postgres database cluster by running the following commands listed below:

[source,bash]
----
pg_lsclusters
----

Drop the existing database cluster, and replace it with a cluster with the locale `en_US.UTF-8`:

[source,bash]
----
sudo pg_dropcluster --stop 10 main
sudo pg_createcluster --locale en_US.UTF-8 --start 10 main
----

Create a new database user called `lisk` and grant it rights to create databases.
Then create the database with the `lisk` user as owner.

> In the last step, define the password for the `lisk` user:

Reviewer: ```suggestion
Finally, define the password for the `lisk` user:
```
